### PR TITLE
Add a usage notification 

### PR DESCRIPTION
**Pull request by dlsniper**
This adds a notification on first run in order to help the user to setup a new project using the plugin.

### CODE REVIEW

**Comment by dlsniper** [comment]
Currently the notification is shown on every project start (multiple times) as the settings part is not properly loading / saving.
Ideally it should be shown on the first start of IDEA after the plugin is installed.

@ignatov or @zolotov could you please point me in the right direction if it's possible to show the notification after IDEA starts, regardless if a project is loaded or not? Thank you!

**Comment by dlsniper** [comment]
I've done the work to make the dialog show only once per session depending on the plugin version, which I hope it's clean enough.

The only question remains, how / if it's possible to show it when IDEA starts instead of when the project is loaded.

Thank you.

File: src/com/goide/settings/GoPluginSettings.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import com.intellij.openapi.components.PersistentStateComponent;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.components.State;
+import com.intellij.openapi.components.Storage;
+import org.jdom.Element;
+import org.jetbrains.annotations.Nullable;
+
+@State(
+  name = "GoPluginSettings",
+  storages = @Storage(id = "other", file = "$APP_CONFIG$/goplugin.xml")
```
Comment (ignatov) [inline src/com/goide/settings/GoPluginSettings.java]: Please include ipr based storage.

Comment (dlsniper) [inline src/com/goide/settings/GoPluginSettings.java]: I'm sorry, I have no idea how to do that. Please advise.

I've tried to add it as it's done for ```GoProjectLibrariesService``` but we don't have it for ```GoApplicationLibrariesService``` and ```GoModuleLibrariesService```.

All I'm getting is:
```
[   9036]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initPersistentComponent(ComponentStoreImpl.java:235)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initComponent(ComponentStoreImpl.java:73)
	at com.intellij.openapi.application.impl.ApplicationImpl.initializeComponent(ApplicationImpl.java:182)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.initializeInstance(ServiceManagerImpl.java:233)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter$1.compute(ServiceManagerImpl.java:216)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:207)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:245)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:211)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:32)
	at com.goide.settings.GoPluginSettings.getInstance(GoPluginSettings.java:45)
	at com.goide.ui.ProjectConfigurationNotification.<init>(ProjectConfigurationNotification.java:44)
[....]
[   9043]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[   9043]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[   9044]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[   9044]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[   9044]  ERROR - impl.stores.ComponentStoreImpl - Current Command: EditorChange 
[   9045]  ERROR - impl.stores.ComponentStoreImpl - Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$ 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initPersistentComponent(ComponentStoreImpl.java:237)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.initComponent(ComponentStoreImpl.java:73)
	at com.intellij.openapi.application.impl.ApplicationImpl.initializeComponent(ApplicationImpl.java:182)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.initializeInstance(ServiceManagerImpl.java:233)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter$1.compute(ServiceManagerImpl.java:216)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:890)
	at com.intellij.openapi.components.impl.ServiceManagerImpl$MyComponentAdapter.getComponentInstance(ServiceManagerImpl.java:207)
	at com.intellij.util.pico.DefaultPicoContainer.getLocalInstance(DefaultPicoContainer.java:245)
	at com.intellij.util.pico.DefaultPicoContainer.getComponentInstance(DefaultPicoContainer.java:211)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:32)
	at com.goide.settings.GoPluginSettings.getInstance(GoPluginSettings.java:45)
	at com.goide.ui.ProjectConfigurationNotification.<init>(ProjectConfigurationNotification.java:44)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[....]
[   9047]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[   9047]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  10124]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:144)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
	at com.intellij.ide.SaveAndSyncHandlerImpl$4.onFrameDeactivated(SaveAndSyncHandlerImpl.java:109)
[...]
[  10125]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[  10125]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[  10126]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[  10126]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  10126]   WARN - mponents.impl.stores.StoreUtil - Save settings failed 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl$StateStorageManagerExternalizationSession.setState(StateStorageManagerImpl.java:303)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:145)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
[....]
[  48807]  ERROR - impl.stores.ComponentStoreImpl - State chooser not specified for: class com.goide.settings.GoPluginSettings 
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:115)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.getComponentStorageSpecs(ComponentStoreImpl.java:309)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:144)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
	at com.intellij.openapi.application.impl.ApplicationImpl.saveSettings(ApplicationImpl.java:1370)
	at com.intellij.ide.SaveAndSyncHandlerImpl.saveProjectsAndDocuments(SaveAndSyncHandlerImpl.java:166)
	at com.intellij.ide.SaveAndSyncHandlerImpl$4.onFrameDeactivated(SaveAndSyncHandlerImpl.java:109)
	at com.intellij.ide.FrameStateManagerImpl.fireDeactivationEvent(FrameStateManagerImpl.java:87)
[....]
[  48808]  ERROR - impl.stores.ComponentStoreImpl - IntelliJ IDEA (Community Edition) IC-140.SNAPSHOT  Build #IC-140.SNAPSHOT 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - JDK: 1.8.0_31 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - VM: Java HotSpot(TM) 64-Bit Server VM 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - Vendor: Oracle Corporation 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - OS: Linux 
[  48808]  ERROR - impl.stores.ComponentStoreImpl - Last Action:  
[  48808]   WARN - mponents.impl.stores.StoreUtil - Save settings failed 
java.lang.IllegalArgumentException: Unknown macro: $PROJECT_FILE$ in storage file spec: $PROJECT_FILE$
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.expandMacros(StateStorageManagerImpl.java:270)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createFileStateStorage(StateStorageManagerImpl.java:210)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.createStateStorage(StateStorageManagerImpl.java:184)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl.getStateStorage(StateStorageManagerImpl.java:101)
	at com.intellij.openapi.components.impl.stores.StateStorageManagerImpl$StateStorageManagerExternalizationSession.setState(StateStorageManagerImpl.java:303)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.commitPersistentComponent(ComponentStoreImpl.java:145)
	at com.intellij.openapi.components.impl.stores.ComponentStoreImpl.save(ComponentStoreImpl.java:102)
	at com.intellij.openapi.components.impl.stores.StoreUtil.save(StoreUtil.java:49)
	at com.intellij.openapi.application.impl.ApplicationImpl._saveSettings(ApplicationImpl.java:1359)
[....]
```

**Comment by dlsniper** [comment]
I've added a commit, https://github.com/dlsniper/google-go-lang-idea-plugin/commit/1093e1df4b21a062a00d687396cae6f27679f1fd, with the changes that I've done in order get the errors I've mentioned in https://github.com/go-lang-plugin-org/go-lang-idea-plugin/pull/1357#discussion_r25137608

If you decide to merge this please note that it will issues those errors for the users.

**Comment by dlsniper** [comment]
I've finally understood what I was doing wrong. I've reread [the docs](https://confluence.jetbrains.com/display/IDEADEV/Persisting+State+of+Components) from Confluence and I've realized that I can't combine a per app setting storage with a per file one.
As I wanted to store the thing on a per app not per project, what I was trying to do made no sense.

It's fixed now and can be merged if there are no additional fixes I need to do. 

Thank you.

File: src/com/goide/settings/GoPluginSettings.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import com.goide.util.GoUtil;
+import com.intellij.openapi.components.*;
+import com.intellij.openapi.roots.impl.ModuleRootManagerComponent;
+import org.jdom.Element;
+import org.jetbrains.annotations.Nullable;
+
+@State(
+  name = GoConstants.GO_SERVICE_NAME,
+  reloadable = true,
+  storages = {
```
Comment (zolotov) [inline src/com/goide/settings/GoPluginSettings.java]: You define it as an application-service but store state in project-file...

File: src/com/goide/settings/GoPluginSettings.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    @Storage(id = "other", file = StoragePathMacros.APP_CONFIG + "/" + GoConstants.GO_CONFIG_FILE, roamingType = RoamingType.PER_PLATFORM)
+  }
+)
+public class GoPluginSettings implements PersistentStateComponent<Element> {
```
Comment (zolotov) [inline src/com/goide/settings/GoPluginSettings.java]: You can just use existing PropertiesComponent for flags like projectTutorialShown. No need create new service for that.

File: src/com/goide/ui/ProjectConfigurationNotification.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      "Please visit our " +
+      "<a href=\"https://github.com/go-lang-plugin-org/go-lang-idea-plugin/wiki/v1.0.0-Setup-initial-project\">wiki page<a/>" +
+      " to learn how to setup a new Go project",
+      NotificationType.INFORMATION,
+      NotificationListener.URL_OPENING_LISTENER
+    );
+    Notifications.Bus.notify(notification);
+    mySettings.setProjectTutorialShown();
+
+    return null;
```
Comment (zolotov) [inline src/com/goide/ui/ProjectConfigurationNotification.java]: EditorNotificationService should create editor notifications (it's panel on the top in editor like 'Setup SDK' one). And it's definitely shouldn't be used for showing Application Notifications only.

**Comment by zolotov** [comment]
> Ideally it should be shown on the first start of IDEA after the plugin is installed.

Use ApplicationComponent for showing notification and PropertiesComponent for storing flags.

File: src/com/goide/ui/ProjectConfigurationNotification.java (hunk 4 above)
Comment (zolotov) [inline src/com/goide/ui/ProjectConfigurationNotification.java]: Also use more descriptive name for notification groups, "GO_PLUGIN" is not good name for showing in UI (see Settings | Appearance | Notifications).
Btw, why did you need create NotificationGroup explicitly?
